safe/impact_function_v4/impact_function.py

```diff
@@ -1337,3 +1356,121 @@ def style(self):
         # Let's style the aggregation and analysis layer.
         simple_polygon_without_brush(self.aggregation_impacted)
         simple_polygon_without_brush(self.analysis_impacted)
+
+    @property
+    def provenance(self):
+        """Helper method to gather provenance for exposure_impacted layer.
+
+        :returns: Dictionary that contains all provenance.
+        :rtype: dict
+        """
+        # Environment
+        self._provenance['host_name'] = gethostname()
```

Comment:
I think we should make compulsory to run `prepare` before calling this property, no ?
Something like https://github.com/inasafe/inasafe/blob/develop/safe/impact_function_v4/impact_function.py#L816 ?